input/docs/handbook/testing/index.md

```diff
@@ -1,28 +1,108 @@
-# Testing
-
-ReactiveUI includes a few tools to help testing, building on what Rx already includes (see [introtorx](http://www.introtorx.com/content/v1.0.10621.0/16_TestingRx.html)) for details.
-
-The utilities are included in the ReactiveUI.Testing nuget package.
+ReactiveUI includes a few tools to help testing, built on what Reactive Extensions for .NET already include. The utilities are included in the `ReactiveUI.Testing` NuGet package. Make sure to install it into your unit tests project.
 
 ## Custom Scheduler
 
-A common pattern is to replace the schedulers ReactiveUI is using to test execution is occuring without injecting schedulers everywhere. This can be achieved in a couple of ways
+> Scheduling, and therefore threading, is generally avoided in test scenarios as it can introduce race conditions which may lead to non-deterministic tests — [Intro to Rx: Testing Rx](http://introtorx.com/Content/v1.0.10621.0/16_TestingRx.html)
+
+By default, `ReactiveCommand` uses `RxApp.MainThreadScheduler` and `ObservableAsPropertyHelper` uses `CurrentThreadScheduler.Instance`, but you can easily override this behavior:
 
-```csharp
-using(TestUtils.WithScheduler(ImmediateScheduler.Instance))
+```cs
+// Inject custom schedulers into your view model.
+public LoginViewModel(IScheduler customScheduler)
 {
-  // your test here
+  // Initialize a reactive command with a custom output scheduler.
+  _loginCommand = ReactiveCommand.CreateFromTask(
+    async () => { /* login logic */ },
+    outputScheduler: customScheduler
+  );
+
+  // Initialize an OAPH with a custom output scheduler.
+  _errorMessage = _loginCommand.ThrownExceptions
+    .Select(exception => exception.Message)
+    .ToProperty(this, x => x.ErrorMessage, scheduler: customScheduler);
 }
 ```
 
-Or
-```csharp
-(new TestScheduler()).With(sched => {
-  ... // Do something
-  sched.AdvanceByMs(2 * 100); // Play with time
-  ...
+This allows you to inject custom schedulers via the constructor of your view model. 
+
+```cs
+// Regular execution.
+var model = new LoginViewModel(RxApp.MainThreadScheduler);
+```
+
+## Unit Tests
+
+Then, in unit tests project, you can inject a `TestScheduler` instance that allows you to play with time:
```

Comment:
Worth noting that the RxApp.MainThreadScheduler (and the other one forgot its name) get changed to TestSchedulers during unit test mode.

Reply:
That is the main reason why those two properties exist to make testing easier.

Reply:
> Worth noting that the RxApp.MainThreadScheduler (and the other one forgot its name) get changed to TestSchedulers during unit test mode.

This is true for commands but false for OAPHs. OAPHs use CurrentThreadScheduler.Instance. Don't know if this is a bug or a feature, but isn't obvious and causes issues with tests :)

Reply:
:) Actually what I've said holds up actually, since it's nothing about ReactiveCommand just the RxApp's schedulers. 

CurrentThread can make sense for some platforms outside WPF etc I guess. 

You potentially may want to use the Rxapp.MainThreadScheduler in your OAPH depending on platform so you don't have to worry about injecting in test schedulers.

Reply:
Will add notes considering this, thanks! 😎 👍